recipes/newrelic/infrastructure/super-agent/rhel.yml

```diff
@@ -0,0 +1,327 @@
+        - task: restart_super_agent
+        - task: assert_super_agent_started
+        - task: post_install
+
+    write_recipe_metadata:
+      cmds:
+        - |
+          echo '{"Metadata":{"CapturedCliOutput":"true"}}' | tee {{.NR_CLI_OUTPUT}} > /dev/null
+
+    assert_pre_req:
+      cmds:
+        - |
+          IS_ROOT_OR_SUDO=$([ ! -z "$SUDO_USER" ] || [ $(whoami) = "root" ] && echo "true" || echo "false") # Assert root or sudo user
+          if [ "$IS_ROOT_OR_SUDO" = "false" ]; then
+            echo "This newrelic install must be run under sudo or root" >&2
+            exit 3
+          fi      
+        - |
+          IS_GREP_INSTALLED=$(which grep | wc -l)
+          if [ $IS_GREP_INSTALLED -eq 0 ] ; then
+            echo "grep is required to run the newrelic install. Please install grep and re-run the installation." >&2
+            exit 10
+          fi
+        - |
+          IS_SED_INSTALLED=$(which sed | wc -l)
+          if [ $IS_SED_INSTALLED -eq 0 ] ; then
+            echo "sed is required to run the newrelic install. Please install sed and re-run the installation." >&2
+            exit 11
+          fi
+        - |
+          IS_AWK_INSTALLED=$(which awk | wc -l)
+          if [ $IS_AWK_INSTALLED -eq 0 ] ; then
+            echo "awk is required to run the newrelic install. Please install awk and re-run the installation." >&2
+            exit 12
+          fi
+        - |
+          IS_CAT_INSTALLED=$(which cat | wc -l)
+          if [ $IS_CAT_INSTALLED -eq 0 ] ; then
+            echo "cat is required to run the newrelic install. Please install cat and re-run the installation." >&2
+            exit 13
+          fi
+        - |
+          IS_TEE_INSTALLED=$(which tee | wc -l)
+          if [ $IS_TEE_INSTALLED -eq 0 ] ; then
+            echo "tee is required to run the newrelic install. Please install tee and re-run the installation." >&2
+            exit 14
+          fi
+        - |
+          IS_TOUCH_INSTALLED=$(which touch | wc -l)
+          if [ $IS_TOUCH_INSTALLED -eq 0 ] ; then
+            echo "touch is required to run the newrelic install. Please install touch and re-run the installation." >&2
+            exit 15
+          fi
+        - |
+          IS_INFRA_AVAILABLE=$(curl -Is {{.NEW_RELIC_DOWNLOAD_URL}}preview/linux/yum/el/{{.DISTRO_VERSION}}/x86_64/newrelic-infra.repo | grep " 2[0-9][0-9] " | wc -l)
+          if [ $IS_INFRA_AVAILABLE -eq 0 ] ; then
+            echo "there is no newrelic super agent available for the distribution '{{.DISTRO_VERSION}}'." >&2
+            exit 21
+          fi
+      vars:
+        DISTRO_VERSION:
+          sh: |
+            rpm -E %{rhel}
+
+    cleanup:
+      cmds:
+        - |
+          if [ "{{.NR_CLI_INFRA_AGENT}}" != "false" ] ; then
+            rm -rf /var/db/newrelic-infra/data 2>/dev/null
+          fi
+
+    setup_infra_license:
+      cmds:
+        - |
+          if [ "{{.NR_CLI_INFRA_AGENT}}" != "false" ] ; then
```

Comment:
What is the intend for {{.NR_CLI_INFRA_AGENT}} variable, and how is it being passed in?

Reply:
NR_CLI_NRDOT --> default will be treated as it was true --> Sets nrdot as enabled and running from the superagent
NR_CLI_INFRA_AGENT --> default will be treated as true --> Sets the infrastrucutre agent as enabled and running from the superagent
NR_CLI_HOST_MONITORING_SOURCE --> default will be treated as not otel (newrelic) --> Makes the host metrics be taken from infra-agent or otel collector.

Those env variables are pased before executing the install from the cli and are going to be set from the UI depending on the selection by the user.

Producing something like:

`https://download.newrelic.com/install/newrelic-cli/scripts/install.sh | bash && sudo  NEW_RELIC_API_KEY=xx NEW_RELIC_ACCOUNT_ID=x NR_CLI_INFRA_AGENT=true NR_CLI_NRDOT=false NR_CLI_HOST_MONITORING_SOURCE=newrelic /usr/local/bin/newrelic install -c path/super/debian.yaml`